rmf_task_ros2/src/rmf_task_ros2/bidding/internal_Auctioneer.hpp

```diff
@@ -34,7 +34,15 @@ class Auctioneer::Implementation
 {
 public:
 
-  std::shared_ptr<rclcpp::Node> node;
+  rclcpp::node_interfaces::NodeBaseInterface::SharedPtr node_base_interface;
+  rclcpp::node_interfaces::NodeClockInterface::SharedPtr node_clock_interface;
+  rclcpp::node_interfaces::NodeLoggingInterface::SharedPtr
+    node_logging_interface;
+  rclcpp::node_interfaces::NodeTimersInterface::SharedPtr node_timers_interface;
+  rclcpp::node_interfaces::NodeTopicsInterface::SharedPtr node_topics_interface;
+  rclcpp::node_interfaces::NodeParametersInterface::SharedPtr
+  node_parameters_interface;
```

Comment:
We could just move the node_interfaces here and call the getters whenever we need access to them? (save memory footprint)

```suggestion
  ActioneerInterfaces node_interfaces;
```